doc/user/unit-testing.md

```diff
@@ -1,132 +1,258 @@
+- [How to increase the verbosity of the logs?](#how-to-increase-the-verbosity-of-the-logs)
 
-`controller-runtime`'s fake client exposes the same set of operations as a typical client, but simply tracks objects rather than sending requests over a network. You can create a new fake client that tracks an initial set of objects with the following code:
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
-```Go
-import (
-  "testing"
+## Overview
+
+Testing your operator should involve both unit and [end-to-end][doc-e2e-test] tests. Unit tests assess the expected outcomes of individual operator components without requiring coordination between components. Operator unit tests should test multiple scenarios likely to be encountered by your custom operator logic at runtime. Much of your custom logic will involve API server calls via a [client][doc-client]; `Reconcile()` in particular will be making API calls on each reconciliation loop. These API calls can be mocked by using `controller-runtime`'s [fake client][doc-cr-fake-client], perfect for unit testing. This document steps through writing a unit test for the [memcached-operator][repo-memcached-reconcile]'s `Reconcile()` method using a fake client.
+
+## Using a Fake client
 
-  cachev1alpha1 "github.com/example-inc/memcached-operator/pkg/apis/cache/v1alpha1"
+The `controller-runtime`'s fake client exposes the same set of operations as a typical client, but simply tracks objects rather than sending requests over a network. You can create a new fake client that tracks an initial set of objects with the following code:
 
-  "k8s.io/apimachinery/pkg/runtime"
-  "sigs.k8s.io/controller-runtime/pkg/client/fake"
+```Go
+import (
+    "testing"
+    cachev1alpha1 "github.com/example-inc/memcached-operator/pkg/apis/cache/v1alpha1"
+    "k8s.io/apimachinery/pkg/runtime"
+    "sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
 func TestMemcachedController(t *testing.T) {
+    ...
+    // A Memcached object with metadata and spec.
+    memcached := &cachev1alpha1.Memcached{
+        ObjectMeta: metav1.ObjectMeta{
+            Name:      "memcached",
+            Namespace: "memcached-operator",
+        },
+    }
 
-  // A Memcached object with metadata and spec.
-  memcached := &cachev1alpha1.Memcached{
-    ObjectMeta: metav1.ObjectMeta{
-      Name:      "memcached",
-      Namespace: "memcached-operator",
-    },
-  }
-  objs := []runtime.Object{memcached}
-  cl := fake.NewFakeClient(objs...)
-
-  ...
-}
+    // Objects to track in the fake client.
+    objs := []runtime.Object{memcached}
 
+    // Create a fake client to mock API calls.
+    cl := fake.NewFakeClient(objs...)
+    ...
+}
 ```
 
-`cl` will cache `memcached` in an internal object tracker so that CRUD operations via `cl` can be performed on it.
+The `cl` will cache `memcached` in an internal object tracker so that CRUD operations via `cl` can be performed on it.
```

Comment:
```suggestion
The fake client `cl` will cache `memcached` in an internal object tracker so that CRUD operations via `cl` can be performed on it.
```

Reply:
👍 done.